src/App.tsx

```diff
@@ -1,28 +1,75 @@
-import { useState } from "react";
-import { getPublicKey } from "./tauriCommands";
+import { useEffect, useRef, useState } from "react";
+
+import { Button } from "./components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "./components/ui/dialog";
+import { handleSignEventRequests } from "./tauriCommands";
 
 const App = () => {
   const [publicKey, setPublicKey] = useState("");
+  const [open, setOpen] = useState(false);
+  const [event, setEvent] = useState({});
```

Comment:
Can we set the type here to the existing `UnsignedNostrEvent` interface rather than an untyped object?